core/src/main/java/io/grpc/internal/TransportSet.java

```diff
@@ -280,16 +271,39 @@ public ClientTransport get() {
       }
     }
 
-    reconnectTask = scheduledExecutor.schedule(
-        new LogExceptionRunnable(new EndOfCurrentBackoff()), delayMillis, TimeUnit.MILLISECONDS);
     return new Runnable() {
       @Override
       public void run() {
+
         // This must be run outside of lock. The TransportSet lock is a channel level lock.
         // startBackoff() will acquire the delayed transport lock, which is a transport level
         // lock. Our lock ordering mandates transport lock > channel lock.  Otherwise a deadlock
         // could happen (https://github.com/grpc/grpc-java/issues/2152).
         delayedTransport.startBackoff(status);
+
+        synchronized (lock) {
+          if (shutdown) {
+            return;
+          }
+          scheduleEndOfCurrentBackoff();
```

Comment:
Are you trying to fix the issue? This should have been mentioned in the PR description.